amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/dynamodb/DynamoDBLeaseManagementFactory.java

```diff
@@ -417,19 +424,23 @@ private DynamoDBLeaseManagementFactory(final KinesisAsyncClient kinesisClient, f
      * @param dynamoDbRequestTimeout
      * @param billingMode
      * @param leaseSerializer
+     * @param leaseCleanupIntervalMillis
+     * @param completedLeaseCleanupThresholdMillis
+     * @param garbageLeaseCleanupThresholdMillis
      */
     public DynamoDBLeaseManagementFactory(final KinesisAsyncClient kinesisClient,
-            final DynamoDbAsyncClient dynamoDBClient, final String tableName, final String workerIdentifier,
-            final ExecutorService executorService, final long failoverTimeMillis, final long epsilonMillis,
-            final int maxLeasesForWorker, final int maxLeasesToStealAtOneTime, final int maxLeaseRenewalThreads,
-            final boolean cleanupLeasesUponShardCompletion, final boolean ignoreUnexpectedChildShards,
-            final long shardSyncIntervalMillis, final boolean consistentReads, final long listShardsBackoffTimeMillis,
-            final int maxListShardsRetryAttempts, final int maxCacheMissesBeforeReload,
-            final long listShardsCacheAllowedAgeInSeconds, final int cacheMissWarningModulus,
-            final long initialLeaseTableReadCapacity, final long initialLeaseTableWriteCapacity,
-            final HierarchicalShardSyncer hierarchicalShardSyncer, final TableCreatorCallback tableCreatorCallback,
-            Duration dynamoDbRequestTimeout, BillingMode billingMode, LeaseSerializer leaseSerializer,
-            Function<StreamConfig, ShardDetector> customShardDetectorProvider) {
+                                          final DynamoDbAsyncClient dynamoDBClient, final String tableName, final String workerIdentifier,
+                                          final ExecutorService executorService, final long failoverTimeMillis, final long epsilonMillis,
+                                          final int maxLeasesForWorker, final int maxLeasesToStealAtOneTime, final int maxLeaseRenewalThreads,
+                                          final boolean cleanupLeasesUponShardCompletion, final boolean ignoreUnexpectedChildShards,
+                                          final long shardSyncIntervalMillis, final boolean consistentReads, final long listShardsBackoffTimeMillis,
+                                          final int maxListShardsRetryAttempts, final int maxCacheMissesBeforeReload,
+                                          final long listShardsCacheAllowedAgeInSeconds, final int cacheMissWarningModulus,
+                                          final long initialLeaseTableReadCapacity, final long initialLeaseTableWriteCapacity,
+                                          final HierarchicalShardSyncer hierarchicalShardSyncer, final TableCreatorCallback tableCreatorCallback,
+                                          Duration dynamoDbRequestTimeout, BillingMode billingMode, LeaseSerializer leaseSerializer,
+                                          Function<StreamConfig, ShardDetector> customShardDetectorProvider, long leaseCleanupIntervalMillis,
```

Comment:
Can we group all the 3 cleanup timers in one object?

Reply:
Yeah, moved these thresholds to a config class.